app/models/export_format.rb

```diff
@@ -0,0 +1,99 @@
+
+  belongs_to :nonprofit
+
+  validates :name, presence: true
+  validates :nonprofit_id, presence: true
+
+  validates_with PosgresqlDateFormatValidator, { attribute_name: :date_format }
+
+  validate :valid_custom_columns_and_values?
+
+  after_validation do
+    normalize_custom_columns
+  end
+
+  private
+
+  ALLOWED_COLUMNS_TO_HAVE_NAMES_CUSTOMIZED = [
+    'payments.date',
+    'payments.gross_amount',
+    'payments.fee_total',
+    'payments.net_amount',
+    'payments.kind',
+    'donations.anonymous',
+    'supporters.anonymous',
+    'donations.anonymous OR supporters.anonymous',
+    'campaigns_for_export.name',
+    'campaigns_for_export.id',
+    'campaigns_for_export.creator_email',
+    'campaign_gift_options.name',
+    'events_for_export.name',
+    'payments.id',
+    'offsite_payments.check_number',
+    'donations.comment',
+    'misc_payment_infos.fee_covered'
+  ].freeze
+
+  ALLOWED_COLUMNS_TO_HAVE_VALUES_CUSTOMIZED = [
+    'payments.kind',
+    'donations.designation',
+    'donations.anonymous',
+    'supporters.anonymous',
+    'donations.anonymous OR supporters.anonymous',
+    'donations.comment',
+    'campaigns_for_export.name',
+    'campaign_gift_options.name',
+    'events_for_export.name',
+    'donations.comment',
+    'misc_payment_infos.fee_covered'
+  ].freeze
+
+  private_constant :ALLOWED_COLUMNS_TO_HAVE_NAMES_CUSTOMIZED
+  private_constant :ALLOWED_COLUMNS_TO_HAVE_VALUES_CUSTOMIZED
+
+  def valid_custom_columns_and_values?
+    return if self.custom_columns_and_values.nil?
+    self.custom_columns_and_values.keys.each do |column|
+      if ALLOWED_COLUMNS_TO_HAVE_NAMES_CUSTOMIZED.include? column
+        unless (custom_columns_and_values[column].include? 'custom_name') || (custom_columns_and_values[column].include? 'custom_values')
+          errors.add(:custom_columns_and_values, "you need to include a 'custom_name' or 'custom_values' key to customize #{column} column")
+        end
+        if (!ALLOWED_COLUMNS_TO_HAVE_VALUES_CUSTOMIZED.include? column) && (custom_columns_and_values[column].include? 'custom_values')
+          errors.add(:custom_columns_and_values, "column #{column} can't have its values customized")
+        end
+      else
+        errors.add(:custom_columns_and_values, "column #{column} does not exist or is not available to be customized")
+      end
+    end
+  end
+
+  def normalize_custom_columns
+    self.custom_columns_and_values&.each do |column, customizations|
+      customizations&.each do |customization, customization_subject|
+        if customization == 'custom_name'
+          self.custom_columns_and_values[column]['custom_name'] =
+            insert_trailing_double_quotes(customization_subject)
+        end
+      end
+    end
+
+    if self.custom_columns_and_values&.dig('supporters.anonymous').present?
+      self.custom_columns_and_values['donations.anonymous OR supporters.anonymous'] = self.custom_columns_and_values['supporters.anonymous']
```

Comment:
So, I don't know if this is the best way to handle this. This code is entirely integrated with the specific design of our current query. If we change the query or use a different mechanism going forward, we have to change all of the export formats already saved. I think the ideal mechanism would be to have a standardized set of property fields on entities being exported and then the export code would decide how to apply those changes.

All of that said: I don't think this is a big enough feature to bother with that immediately.

Reply:
I removed that and changed it to be handled in the query itself https://github.com/CommitChange/houdini/pull/130/files#diff-5c62657635181a93da46ee3477e7c82881a24305e549b9f74000fc3ba696339bR218-R222